src/main/java/javaslang/Value.java

```diff
@@ -510,6 +513,12 @@ default CharSeq toCharSeq() {
         return ValueModule.toTraversable(this, Vector.empty(), Vector::of, Vector::ofAll);
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    default Match.MatchMonad.Of<T> match() {
+        return Match.of((T) this);
```

Comment:
There is a reason that `(T) this` does not check. It is `Convertible<T>` and cannot be cast to `T`.

I reopened #968 and already fixed it locally. PR comes soon...